fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java

```diff
@@ -0,0 +1,69 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.plan;
+
+import com.google.common.collect.Lists;
+import com.starrocks.common.FeConstants;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class RemoveAggTest extends PlanTestBase {
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        PlanTestBase.beforeClass();
+        FeConstants.runningUnitTest = true;
+    }
+
+    @ParameterizedTest(name = "sql_{index}: {0}.")
+    @MethodSource("removeAggSqlCases")
+    void removeAggTest(String sql) throws Exception {
+        String plan = getFragmentPlan(sql);
+        assertNotContains(plan, "AGGREGATE");
+        assertContains(plan, "PREAGGREGATION: OFF");
+    }
+
+
+    @ParameterizedTest(name = "sql_{index}: {0}.")
+    @MethodSource("remainAggSqlCases")
+    void remainAggTest(String sql) throws Exception {
+        String plan = getFragmentPlan(sql);
+        assertContains(plan, "AGGREGATE");
+    }
+
+
+    private static Stream<Arguments> removeAggSqlCases() {
+        List<String> sqlList = Lists.newArrayList();
+        sqlList.add("select max(v1) from test_agg group by k1, k2, k3;");
+        sqlList.add("select max(v1) from test_agg group by k1, k2, k3 having(max(v1) = 1)");
+        sqlList.add("select max(v1), min(v2) from test_agg group by k1, k2, k3 " +
+                "having (max(v1) > min(v2) or abs(max(v1)) is null)");
+        sqlList.add("select max(v1), min(v2), abs(123) from test_agg group by k1, k2, k3 having (max(v1) + min(v2) > sum(v3))");
+        sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 having (max(v1) + min(v2)) is not null");
+        sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 " +
+                "having ((max(v1) + min(v2)) is not null or k1 > 4)");
+        sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having k1 > 1");
```

Comment:
I think we should test having filter is remain.

Reply:
Yes. Thess cases are testing when agg is removed but its having filter are transformed to new filter on the child operator.

Reply:
add the predicate check.